projects/wizard_of_wikipedia/wizard_transformer_ranker/wizard_dict.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from parlai.core.build_data import modelzoo_path
+from parlai.core.dict import DictionaryAgent
+
+from collections import defaultdict
+import copy
+import os
+
+import re
+
+RETOK = re.compile(r'\w+|[^\w\s]|\n', re.UNICODE)
+
+
+class WizardDictAgent(DictionaryAgent):
+    def __init__(self, opt, shared=None):
+        # initialize fields
+        self.opt = copy.deepcopy(opt)
+        self.minfreq = opt.get('dict_minfreq', DictionaryAgent.default_minfreq)
+        self.null_token = '__PAD__'
+        self.end_token = '__SOC__'
+        self.unk_token = '__UNK__'
+        self.start_token = '__SOC__'
```

Comment:
start and end are same token? interesting, didn't realize before

Reply:
actually i think there shouldn't be an end token at all

Reply:
nice catch